libbeat/template/template.go

```diff
@@ -193,26 +196,26 @@ func (t *Template) GetPattern() string {
 // The default values are taken from the default variable.
 func (t *Template) Generate(properties common.MapStr, dynamicTemplates []common.MapStr) common.MapStr {
 	keyPattern, patterns := buildPatternSettings(t.esVersion, t.GetPattern())
-
-	return common.MapStr{
+	template := common.MapStr{
 		keyPattern: patterns,
-
-		"mappings": buildMappings(
-			t.beatVersion, t.esVersion, t.beatName,
-			properties,
-			append(dynamicTemplates, buildDynTmpl(t.esVersion)),
-			common.MapStr(t.config.Settings.Source),
-		),
-
-		"order": 1,
-
+		"order":    t.order,
 		"settings": common.MapStr{
 			"index": buildIdxSettings(
 				t.esVersion,
 				t.config.Settings.Index,
 			),
 		},
 	}
+	if len(properties) == 0 && len(dynamicTemplates) == 0 {
+		return template
+	}
```

Comment:
Why introduce this early return here? even if properties and dynamicTemplates is empty, we still have `t.config.Settings.Source` and the call `buildDynTmpl`, which always returns a common.MapStr.

Reply:
That was intended to keep additional templates small, but I see how it can be problematic, changed it back now.